File: typescript/src/models/totalType.ts
// this file is @generated

export enum TotalType {
  ItemsBaseAmount = "items_base_amount",
  ItemsDiscount = "items_discount",
  Subtotal = "subtotal",
  Discount = "discount",
  Fulfillment = "fulfillment",
  Tax = "tax",
  Fee = "fee",
  Total = "total",
}

export const TotalTypeSerializer = {
  _fromJsonObject(object: any): TotalType {
    return object;
  },

  _toJsonObject(self: TotalType): any {
    return self;
  },
};

File: rust/src/models/total_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum TotalType {
    #[serde(rename = "items_base_amount")]
    ItemsBaseAmount,
    #[serde(rename = "items_discount")]
    ItemsDiscount,
    #[serde(rename = "subtotal")]
    Subtotal,
    #[serde(rename = "discount")]
    Discount,
    #[serde(rename = "fulfillment")]
    Fulfillment,
    #[serde(rename = "tax")]
    Tax,
    #[serde(rename = "fee")]
    Fee,
    #[serde(rename = "total")]
    Total,
}

impl fmt::Display for TotalType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::ItemsBaseAmount => "items_base_amount",
            Self::ItemsDiscount => "items_discount",
            Self::Subtotal => "subtotal",
            Self::Discount => "discount",
            Self::Fulfillment => "fulfillment",
            Self::Tax => "tax",
            Self::Fee => "fee",
            Self::Total => "total",
        };
        f.write_str(value)
    }
}
